Removed 1 unnecessary stubbing in GithubSCMSourceBranchesTest.java

ARUS2023 opened the pull request:
# Description

A brief summary describing the changes in this pull request. 

In our analysis of the project, we observed that the test `GithubSCMSourceBranchesTest.testThrownErrorSingleBranchException` contains 1 unnecessary stubbing. 

Unnecessary stubbings are stubbed method calls that were never realized during test execution. Mockito recommends to remove unnecessary stubbings (https://www.javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/exceptions/misusing/UnnecessaryStubbingException.html). 

We propose below a solution to remove the unnecessary stubbing.
<!--
In the lists below, fill in the empty checkboxes [ ] with checks by replacing the space with an x, like [x].
-->
# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Automated tests have been added to exercise the changes
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verify that the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

# Documentation changes
- [ ] Link to jenkins.io PR, or an explanation for why no doc changes are needed

# Users/aliases to notify

jglick reviewed (changes requested):
Something is wrong if this passes, since then https://github.com/jenkinsci/github-branch-source-plugin/blob/a502178d0be118f4023f81b889fee70f87de6b2f/src/test/java/org/jenkinsci/plugins/github_branch_source/GithubSCMSourceBranchesTest.java#L75-L75 is unused and so where is https://github.com/jenkinsci/github-branch-source-plugin/blob/a502178d0be118f4023f81b889fee70f87de6b2f/src/test/java/org/jenkinsci/plugins/github_branch_source/GithubSCMSourceBranchesTest.java#L89 coming from? …because the assertion was wrong; it should have been using `error`, not `e`!

ARUS2023 commented:
Hello,
Thanks for the reminder. We changed the `e` to `error` at this line. 
https://github.com/jenkinsci/github-branch-source-plugin/blob/a502178d0be118f4023f81b889fee70f87de6b2f/src/test/java/org/jenkinsci/plugins/github_branch_source/GithubSCMSourceBranchesTest.java#L89

We also found out that the stubbing is not executed because the stubbed method should be `getBranch` instead of `getRef`.

We have updated the solution.

Thanks.
